### PR TITLE
Fix handling of grid node id

### PR DESCRIPTION
**Pull request by tommywo**
## Proposed changes

This fixes issue when user specifies id of grid node,
but appium keep using default id.
## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._
- [x] I have read the [CONTRIBUTING](/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://docs.google.com/forms/d/1lOfXRw_0VCk7gYzjj4WLetGu7yelDVo5LWh0z7pGftE/viewform)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by triager** [comment]
Can one of the admins verify this patch?

**Comment by triager** [comment]
Can one of the admins verify this patch?

**Comment by tommywo** [comment]
Travis build failure doesn't seems to  be related to pull request.

```
npm ERR! notsup Not compatible with your operating system or architecture: node-ios-device@0.9.5
npm ERR! notsup Valid OS:    darwin
npm ERR! notsup Valid Arch:  any
npm ERR! notsup Actual OS:   linux
npm ERR! notsup Actual Arch: x64
```

File: lib/grid-register.js [hunk 1]
```diff
@@ -55,6 +55,13 @@ async function postRequest (data, addr, port) {
     // re-serialize the configuration with the auto populated data
     data = JSON.stringify(jsonObject);
   }
+  // if the node config does not have id automatically add it
+  if (!jsonObject.configuration.id) {
+    jsonObject.configuration.id = `http://${jsonObject.configuration.host}:${jsonObject.configuration.port}`;
+
+    // re-serialize the configuration with the auto populated data
+    data = JSON.stringify(jsonObject);
```
Comment (imurchie) [inline lib/grid-register.js]: This, and the one above, should be moved out and only done once, so we end up with

```js
if (!jsonObject.configuration.url || !jsonObject.configuration.host || !jsonObject.configuration.port) {
  jsonObject.configuration.url = `http://${addr}:${port}/wd/hub`;
  jsonObject.configuration.host = addr;
  jsonObject.configuration.port = port;
}
// if the node config does not have id automatically add it
if (!jsonObject.configuration.id) {
  jsonObject.configuration.id = `http://${jsonObject.configuration.host}:${jsonObject.configuration.port}`;
}

// re-serialize the configuration with the populated data
data = JSON.stringify(jsonObject);
```

There is little overhead to stringifying the JSON object.

**Comment by imurchie** [comment]
Yup. That is a problem elsewhere. I'll fix it.

**Comment by imurchie** [comment]
Thanks for this! One small comment to reduce a little redundancy, but otherwise great!

**Comment by tommywo** [comment]
I've updated the code according to your comment. Looking forward to having this released! :)

**Comment by imurchie** [comment]
I'm just fixing the issue that is making install on Linux break. Shouldn't be too long.

**Comment by imurchie** [comment]
Finally fixed the issue (which turned out to be two dependencies). 

Thanks for getting this in!

**Comment by jlipps** [comment]
thanks @tommywo!

**Pull request merged**